feat(angular): Add Angular 20 compatibility

thakkarkeyur opened the pull request:
## Summary
Adds Angular 20 compatibility to dockview-angular while maintaining backward compatibility with Angular 14+.

## Problem
Angular 20 introduced stricter validation for standalone components. The current implementation causes compilation errors:
- NG6008: Component is standalone and cannot be declared in an NgModule
- NG6004: Can't be exported from NgModule without being imported first

## Solution
Implements a hybrid standalone pattern:
- Added `standalone: true` to all four components (DockviewAngular, GridviewAngular, PaneviewAngular, SplitviewAngular)
- Updated `DockviewAngularModule` to import components instead of declaring them
- This pattern works with both NgModule and standalone component usage

## Changes
- `dockview-angular.component.ts` - Added standalone: true
- `gridview-angular.component.ts` - Added standalone: true
- `paneview-angular.component.ts` - Added standalone: true
- `splitview-angular.component.ts` - Added standalone: true
- `dockview-angular.module.ts` - Changed from declarations to imports

## Testing
- Verified compilation with Angular 20.3.12 and TypeScript 5.8
- Tested with minimal Angular 20 application
- Confirmed backward compatibility maintained (Angular 14-19)
- Zero breaking changes for existing users

## Breaking Changes
None - fully backward compatible.

fintech-keyur-thakkar commented:
## 🔧 Critical Bug Fix Added

I've added an important bug fix to this PR that addresses a fundamental issue with the AngularRenderer.

### Problem
The `AngularRenderer` was using `hasOwnProperty()` to check if properties exist before setting them. This doesn't work with TypeScript getter/setter properties because they're defined on the prototype chain, not as own properties.

This bug prevented header action components (and any components using getters/setters) from receiving their props from dockview-core.

### Solution
**Commit: `842ca163d`**

1. **Changed property detection**: Replaced `hasOwnProperty()` with the `in` operator to check the entire prototype chain
2. **Made `init()` idempotent**: Updated init method to handle multiple initialization calls
3. **Added documentation**: Clarified the initialization pattern in component-factory.ts

### Files Changed
- `packages/dockview-angular/src/lib/utils/angular-renderer.ts`
- `packages/dockview-angular/src/lib/utils/component-factory.ts`

### Impact
This enables reactive patterns with property setters:
```typescript
export class HeaderActionsComponent {
    private _api: any;
    
    get api() { return this._api; }
    set api(value: any) {
        this._api = value;
        this.setupListeners();  // Now this works!
    }
}
```

### Testing
- ✅ Verified with Angular 20.3.12
- ✅ Confirmed header action components receive props correctly
- ✅ Tested getter/setter properties work as expected
- ✅ Backward compatible with regular properties

This is a general bug fix that benefits all Angular versions, not just Angular 20.

thakkarkeyur commented:
## Critical Bug Fix Added

Adds an important bug fix to this PR that addresses a fundamental issue with the AngularRenderer.

### Problem
The AngularRenderer was using `hasOwnProperty()` to check if properties exist before setting them. This doesn't work with TypeScript getter/setter properties because they're defined on the prototype chain, not as own properties.

This bug prevented header action components (and any components using getters/setters) from receiving their props from `dockview-core`.

### Solution
**Commit:**  https://github.com/mathuo/dockview/pull/1036/commits/842ca163dbc14d009fa106f319d22c08f1533103

- Changed property detection: Replaced `hasOwnProperty()` with the `in` operator to check the entire prototype chain
- Made `init()` idempotent: Updated `init` method to handle multiple initialization calls
- Added documentation: Clarified the initialization pattern in `component-factory.ts`

### Files Changed
- `packages/dockview-angular/src/lib/utils/angular-renderer.ts`
- `packages/dockview-angular/src/lib/utils/component-factory.ts`

### Impact
This enables reactive patterns with property setters:

```ts
export class HeaderActionsComponent {
    private _api: any;
    
    get api() { return this._api; }
    set api(value: any) {
        this._api = value;
        this.setupListeners();  // Now this works!
    }
}
````

### Testing

* Verified with Angular 20.3.12
* Confirmed header action components receive props correctly
* Tested getter/setter properties work as expected
* Backward compatible with regular properties

This is a general bug fix that benefits all Angular versions, not just Angular 20.
